pkg/service/rtcservice.go

```diff
@@ -265,17 +286,24 @@ func (s *RTCService) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 					continue
 				}
 
-				if err = sigConn.WriteResponse(res); err != nil {
+				if pi.ID == "" && initialResponse.GetJoin() != nil {
+					pi.ID = livekit.ParticipantID(initialResponse.GetJoin().GetParticipant().GetSid())
+					signalStats = telemetry.NewBytesTrackStats(false, pi.ID, s.telemetry)
```

Comment:
should this be creating a new object only if not nil?

Reply:
The object will not be nil if pi.ID is not empty